events/story_cycles/wc_story_cycle_horde_invasion_events.txt

```diff
@@ -47,39 +47,150 @@ WCHOI.2 = {
 		}
-		name = WCHOI_2_OPT_A
 	}
+}
 
-	option = { # Frostwolf option
+# Ends the Horde Invasion when enough lands taken
+WCHOI.3 = { # Save the Horde?
+	type = character_event
+	title = WCHOI_2_TITLE
+	desc = WCHOI_3_DESC
+	theme = war
+	override_background = {
+		reference = wc_background_dark_portal
+	}
+	left_portrait = root
+
+	option = { # The orcs need their warchief!
 		trigger = {
-			NOT = { 
-				faith = faith:orcish_fel
+			trigger_if = {
+				limit = { NOT = { has_game_rule = wc_collapse_of_the_first_horde_never } } # If has this gamerule, always available
+
+				trigger_if = {
+					limit = { NOT = { has_game_rule = wc_collapse_of_the_first_horde_always } }
+
+					prestige_level >= max_prestige_level
+				}
+				trigger_else = { always = no }
+			}
+		}
+		show_as_unavailable = {
+			NOT = { has_game_rule = wc_collapse_of_the_first_horde_always }
+		}
+
+		name = WCHOI_3_OPT_A
+
+		if = {
+			limit = {
+				NOR = {
+					has_game_rule = wc_collapse_of_the_first_horde_can_collapse_not
+					has_game_rule = wc_collapse_of_the_first_horde_never
+				}
+			}
+
+			add_tyranny = 60
+
+			custom_tooltip = wc_members_choice_the_first_horde_tooltip
+
+			save_scope_as = warchief
+
+			every_vassal = {
+				add_to_temporary_list = horde_vassals
+			}
+
+			every_in_list = {
+				list = horde_vassals
+
+				trigger_event = WCHOI.4
+			}
+		}
+		else = {
+			custom_tooltip = wc_save_the_first_horde_tooltip
+
+			if = {
+				limit = { NOT = { has_game_rule = wc_collapse_of_the_first_horde_never } }
+
+				add_tyranny = 90
+			}
+		}
+
+		stress_impact = {
+			ambitious = medium_stress_impact_loss
+			greedy = medium_stress_impact_loss
+			arbitrary = medium_stress_impact_loss
+			just = medium_stress_impact_gain
+			generous = medium_stress_impact_gain
+			content = medium_stress_impact_gain
+		}
+
+		ai_chance = {
+			base = 100
+			ai_value_modifier = {
+				ai_greed = 1
+				ai_honor = -1
+			}
+			modifier = {
+				add = 200
+
+				has_trait = ambitious
+			}
+			modifier = {	# Weight up for stress.
+				add = 20
+				has_trait = greedy
+			}
+			modifier = {	# Weight up for stress.
+				add = 20
+				has_trait = arbitrary
+			}
+			modifier = {	# Weight up for stress.
+				add = -20
+				has_trait = just
+			}
+			modifier = {	# Weight up for stress.
+				add = -20
+				has_trait = generous
+			}
+			modifier = {	# Weight up for stress.
+				add = -20
+				has_trait = content
 			}
-			faith = faith:orcish_shamanism
 		}
-		name = WCHOI_2_OPT_B
 	}
-	
-	option = { # Outsider option
+	option = { # We'll go our own way
 		trigger = {
-			NOT = { 
-				faith = faith:orcish_shamanism
-				faith = faith:orcish_fel
+			NOT = { has_game_rule = wc_collapse_of_the_first_horde_never }
+		}
+		name = WCHOI_3_OPT_B
+
+		destroy_title_free_vassals_effect = { limit_size = minor_realm_size }
+
+		ai_chance = {
+			base = 200
+			ai_value_modifier = {
+				ai_greed = -1
+				ai_compassion = 1
 			}
 		}
-		name = WCHOI_2_OPT_C
 	}
```

Comment:
In this option, vassals don't get any kind of notification about what happened.

Reply:
Added?